# MyGenome
*The following is a basic record of the methods and results from the whole genome assembly of rice blast fungus, Pyricularia oryzae.*

## Sequence quality assessment
*Upload and assess the quality of the forward and reverse sequences. Trim if adaptor content is above 5%.*
1. Upload sequences as .gz files.
2. Run both forward and reverse sequences through FastQC for quality assessment.
```
fastqc Pd8825_1.fq.gz
fastqc Pd8825_2.fq.gz
```
NOTE: For ease of access, the resulting .html files generated by FastQC can be uploaded to the local computer. From the FastQC, it is clear that the overall quality is high, however, adaptor content is still present (~12%). 
![Sequence Quality Base](https://github.com/user-attachments/assets/439a15a1-7d5e-421c-8b23-422fad38200a)
![Sequence Quality Tile](https://github.com/user-attachments/assets/124917e4-0c8d-4286-be62-97d4e591c8e4)
![Initial Adaptor Content](https://github.com/user-attachments/assets/cb921a02-5f58-43b1-9931-6fe725bc103a)
*Images above are from the forward reads.*

## Trimming
*To remove adaptor content, the following Trimmomatic command-line was used to trim both forward and reverse sequences.*
1. Run customized Trimmomatic command.
```
java -jar ~/sequences/trimmomatic-0.38.jar PE -threads 2 -phred33 -trimlog Pd8825_errorlog.txt Pd8825_1.fq Pd8825_2.fq Pd8825_1_paired.fastq Pd8825_1_unpaired.fastq Pd8825_2_paired.fastq Pd8825_2_unpaired.fastq ILLUMINACLIP:adaptors.fasta:2:30:10 SLIDINGWINDOW:20:20 MINLEN:140
```
2. Run paired sequences through FASTQC to assess the quality of the trimming.
```
fastqc Pd8825_1_paired.fastq
fastqc Pd8825_2_paired.fastq
```
*Another FastQC revealed the adaptor content reduced to ~2% in the resulting trimmed sequences.*
![Base](https://github.com/user-attachments/assets/81f0997b-49c6-41bc-a33a-2267055d3077)
![tile](https://github.com/user-attachments/assets/05782ad3-2267-4a01-afd1-981113c4d759)
![adaptor](https://github.com/user-attachments/assets/5f2530df-92b9-40e7-944a-37dc20904694)

The following table summarizes the results from Trimmomatic:

|  Sequence  | Reads | Adaptor Content |
| ------------- | ------------- | -------------- |
| Raw Forward  | 7692782  | ~12% |
| Raw Reverse  | 7692782  | ~12% |
| Trimmed Forward  | 6356740  | ~2% |
| Trimmed Reverse  | 6356740  | ~2% |

***NOTE:** HTML files from the FastQC before and after trimming can be found under the `FastQC` directory.*
## Assembly
Prepare for genome assembly by transferring files from virtual machine to MCC. 
1. Log into MCC.
2. Use scp to transfer trimmed sequence reads from virtual machine to MCC machine.
Determine optimal k-mer value for genome assembly with Velvet.
3. Use [Velvet Advisor](https://dna.med.monash.edu.au/~torsten/velvet_advisor/) to get an initial k-mer value suggestion.
4. Run VelvetOptimiser by steps of 10 to find an approximate optimal k-mer value. *NOTE: We use a batch script to run this command. This batch file can be found under `SLURM_SCRITPS`.*
```
sbatch velvetoptimiser_noclean.sh Pd8825 61 131 10
```
Results from 10-step:
|  k-mer value  | Genome size | # of contigs | N50 | Fold coverage |
| ------------- | ------------- | -------------- | -------------- |-------------- |
| 91 | 42,162,223  | 5,418 | 34,028 | 45.23 |

5. Run VelvetOptimiser by steps of 2 to find most optimal k-mer value (based on 10-step results).
```
sbatch velvetoptimiser_noclean.sh Pd8825 71 111 2
```
Results from 2-step:
|  k-mer value  | Genome size | # of contigs | N50 | Fold coverage |
| ------------- | ------------- | -------------- | -------------- | -------------- |
| 97 | 42,219,964  | 5,455 | 28,275 | 45.16 |

6. Run Velvet genome assembly with optimal k-mer value.
```
velveth Pd8825_97_2 97 -shortPaired -fastq -separate Pd8825_1_paired.fastq Pd8825_2_paired.fastq
velvetg Pd8825_97_2
```
*Note: The following command is required before running the code above if Velvet is not installed in machine.*
```
singularity exec /share/singularity/images/ccs/conda/amd-conda2-centos8.sinf
```
## Process and finalize assembly
1. Finalize genome assembly (removing short contigs and checking sequence length).
```
perl SimpleFastaHeaders.pl Pd8825_97_2/Pd8825.fasta
perl Pd8825_97/CullShortContigs.pl Pd8825_97_2/Pd8825_nh.fasta
perl Pd8825_97/SeqLen.pl Pd8825_97_2/Pd8825_final.fasta
```
*Note: Scripts can be found under `SCRIPTS`*

2. Assess the completness of the genome assembly with BUSCO.
```
sbatch BuscoSingularity.sh Pd8825_97_2/Pd8825_final.fasta
```
*Note: Scripts can be found under `SLURM_SCRIPTS`*
Results from BUSCO:
|  BUSCO score (complete)  | BUSCO score (complete + fragmented) | 
| ------------- | ------------- | 
| 91.2% | 96.7%  | 

This BUSCO score is extremely low!! This result leads to some concern so we ran some diagnostics *(more information about this at the end)*.

3. Identify mitochondrial genome and export a list of contigs with mitochondrial DNA for NCBI submission.
```
blastn -query MoMitochondrion.fasta -subject Pd8825_97_2/Pd8825_nh.fasta -evalue 1e-50 -max_target_seqs 20000 -outfmt '6 qseqid sseqid slen length qstart qend sstart send btop' -out MoMitochondrion.Pd8825.BLAST
awk '$4/$3 > 0.9 {print $2 ",mitochondrion"}' MoMitochondrion.Pd8825.BLAST > Pd8825_mitochondrion.csv
```
*Note: The following command is required before running the code above if blastn is not installed in machine.*
```
singularity run --app blast2120 /share/singularity/images/ccs/conda/amd-conda1-centos8.sinf
```
## BLAST comparison against reference genome
1. Use blastn to run a BLAST search against the reference genome (B71). Use singularity if needed (see above). 
```
blastn -query B71v2sh_masked.fasta -subject Pd8825_97_2/Pd8825_final.fasta -evalue 1e-50 -max_target_seqs 20000 -outfmt '6 qseqid sseqid qstart qend sstart send btop' -out B71v2sh.Pd8825.BLAST
```

## Gene prediction

## Diagnostics 
1. Examining the deviance from the reference genome within specific contigs in the Velvet assembly using BLAST (format 6).
```
blastn -query B71v2sh_masked.fasta -subject Pd8825_final.fasta -evalue 1e-50 -outfmt 6 -out B71v2sh.Pd8825.BLAST.6
sort -k1,1 -k7n B71v2sh.Pd8825.BLAST.6 | more
```
Singularity:
```
singularity run --app blast2120 /share/singularity/images/ccs/conda/amd-conda1-centos8.sinf 
```
From this BLAST we found that large portions of the genome have low deviation, but there are also substantial portions of high deviation (95-97% with contigs length >20,000 reads). If we wanted to manually identify some of these longer contigs with high deviation, we could run the following command (including singularity):
```
grep -A 1 Pd8825_contig5889 Pd8825_final.fasta | NR=2 awk '{print substr($1,1,14629)}' | singularity run --app blast2120 /share/singularity/images/ccs/conda/amd-conda1-centos8.sinf blastn -query - -db nr -outfmt 6 -remote
```
In this command, the contig we identified was Pd8825_contig5889. This BLAST primarily returned hits to *Pyricularia oryzae* so this command was not very fruitful. Alternatively, we would like to identiy the Internal Transcribed Spacers (ITS) regions to BLAST instead. 

2. Assemble new genome with SPAdes.
The following SPAdes pipline trims, assmebles, and finalizes the genome assmebly. Thus the only input required are the raw reads and bash script.
```
sbatch trim-spades.sh . Pd8825 yes
```
*Note: Bash script can be found under `SLURM_SCRIPTS`*

Results of SPAdes assembly:
|  BUSCO score (complete)  | BUSCO score (complete + fragmented) | 
| ------------- | ------------- | 
| 98.4% | 98.6%  | 

```
blastn -query B71v2sh_masked.fasta -subject spades/Pd8825/Pd8825_final.fasta -evalue 1e-50 -outfmt 6 -out B71v2sh.Pd8825.spades.BLAST.6
sort -k1,1 -k7n B71v2sh.Pd8825.spades.BLAST.6 | more
```
Singularity:
```
singularity run --app blast2120 /share/singularity/images/ccs/conda/amd-conda1-centos8.sinf 
```
3. Velvet assembly with SPAdes' trimmed reads.
4. SPAdes assembly with initial manual trimmed reads. 
