# MyGenome
*The following is a basic record of the methods and results from the whole genome assembly of rice blast fungus, Pyricularia oryzae.*

*Pyricularia oryzae*, now known as *Magnaporthe oryzae* (2002), causes rice blast and is a major plant pathogen infecting rice and other cereal crops such as wheat, rye, and barley. *P. oryzae* was sampled from host plant *Paspalum distichum* and sequenced with the ILLUMINA NovaSeq X platform. 

## Sequence quality assessment
*Upload and assess the quality of the forward and reverse sequences. Trim if adaptor content is above 5%.*
1. Run both forward and reverse sequences through FastQC for quality assessment.
```
fastqc Pd8825_1.fq.gz
fastqc Pd8825_2.fq.gz
```
For ease of access, the resulting .html files generated by FastQC can be uploaded to the local computer. From the FastQC, it is clear that the overall quality is high, however, adaptor content is still present (~12%). 
<details>
<summary>Click on caret to view full FastQC results</summary>
  
![QualityBeforeTrim](https://github.com/user-attachments/assets/b24a961d-90b6-4b0d-980d-5fb7bba3e6c5)

*Images above are from the reverse reads.*
</details>

## Trimming
*To remove adaptor content, the following Trimmomatic command-line was used to trim both forward and reverse sequences.*
1. Run customized Trimmomatic command.
```
java -jar ~/sequences/trimmomatic-0.38.jar PE -threads 2 -phred33 -trimlog Pd8825_errorlog.txt Pd8825_1.fq Pd8825_2.fq Pd8825_1_paired.fastq Pd8825_1_unpaired.fastq Pd8825_2_paired.fastq Pd8825_2_unpaired.fastq ILLUMINACLIP:adaptors.fasta:2:30:10 SLIDINGWINDOW:20:20 MINLEN:140
```
2. Run paired sequences through FASTQC to assess the quality of the trimming.
```
fastqc Pd8825_1_paired.fastq
fastqc Pd8825_2_paired.fastq
```
*Another FastQC revealed the adaptor content reduced to ~2% in the resulting trimmed sequences.*

<details>
<summary>Click on caret to view full FastQC results</summary>

![QualityAfterTrim](https://github.com/user-attachments/assets/373fe055-9f2e-445a-9183-2ee446a9faf4)

</details>

The following table summarizes the results from Trimmomatic:

|  Sequence  | Reads | Adaptor Content |
| ------------- | ------------- | -------------- |
| Raw Forward  | 7692782  | ~12% |
| Raw Reverse  | 7692782  | ~12% |
| Trimmed Forward  | 6356740  | ~2% |
| Trimmed Reverse  | 6356740  | ~2% |

***NOTE:** HTML files from the FastQC before and after trimming can be found under the `FastQC` directory.*
## Assembly
*Previous tasks were completed in the virtual machine. The following assmebly will be completed on MCC.* 
1. Use scp to transfer trimmed sequence reads from virtual machine to MCC machine.
Determine optimal k-mer value for genome assembly with Velvet.
2. Use [Velvet Advisor](https://dna.med.monash.edu.au/~torsten/velvet_advisor/) to get an initial k-mer value suggestion.
3. Run VelvetOptimiser by steps of 10 to find an approximate optimal k-mer value. ***NOTE:** We use a bash script to run this command. This bash file can be found under `SLURM_SCRITPS`.*
```
sbatch velvetoptimiser_noclean.sh Pd8825 61 131 10
```
Results from 10-step:
|  k-mer value  | Genome size | # of contigs | N50 | Fold coverage |
| ------------- | ------------- | -------------- | -------------- |-------------- |
| 91 | 42,162,223  | 5,418 | 34,028 | 45.23 |

4. Run VelvetOptimiser by steps of 2 to find most optimal k-mer value (adjust range based on 10-step results).
```
sbatch velvetoptimiser_noclean.sh Pd8825 71 111 2
```
Results from 2-step:
|  k-mer value  | Genome size | # of contigs | N50 | Fold coverage |
| ------------- | ------------- | -------------- | -------------- | -------------- |
| 97 | 42,219,964  | 5,455 | 28,275 | 45.16 |

5. Run Velvet genome assembly with optimal k-mer value.
```
velveth Pd8825_97_2 97 -shortPaired -fastq -separate Pd8825_1_paired.fastq Pd8825_2_paired.fastq
velvetg Pd8825_97_2
```
***NOTE:** The following command is required before running the code above if Velvet is not installed in machine.*
```
singularity run --app velvet1210 /share/singularity/images/ccs/conda/amd-conda2-centos8.sinf
```
## Process and finalize assembly
*To finalize genome assembly, the following renames contig headers and removes short reads (<200 bp).*
1. Finalize genome assembly.
Before running the following commands, rename contigs.fa to Pd8825.fasta...
```
perl SimpleFastaHeaders.pl Pd8825_97_2/Pd8825.fasta
perl Pd8825_97/CullShortContigs.pl Pd8825_97_2/Pd8825_nh.fasta
perl Pd8825_97/SeqLen.pl Pd8825_97_2/Pd8825_final.fasta
```
***NOTE:** Scripts can be found under `SCRIPTS`*

As the names suggest:
- SimpleFastaHeaders.pl renames the headers (for each contig)
- CullShortContigs.pl removes small contigs
- SeqLen.pl checks overall sequence length and the length of each contig

2. Assess the completness of the genome assembly with BUSCO.
```
sbatch BuscoSingularity.sh Pd8825_97_2/Pd8825_final.fasta
```
***NOTE:** Scripts can be found under `SLURM_SCRIPTS`*
Results from BUSCO:
|  BUSCO score (complete)  | BUSCO score (complete + fragmented) | 
| ------------- | ------------- | 
| 91.2% | 96.7%  | 

This BUSCO score is extremely low!! This result leads to some concern so we ran some diagnostics *(more information about this at the end)*.

3. Identify mitochondrial genome and export a list of contigs with mitochondrial DNA for NCBI submission.
```
blastn -query MoMitochondrion.fasta -subject Pd8825_97_2/Pd8825_final.fasta -evalue 1e-50 -max_target_seqs 20000 -outfmt '6 qseqid sseqid slen length qstart qend sstart send btop' -out MoMitochondrion.Pd8825.BLAST
awk '$4/$3 > 0.9 {print $2 ",mitochondrion"}' MoMitochondrion.Pd8825.BLAST > Pd8825_mitochondrion.csv
```
***NOTE:** The following command is required before running the code above if blastn is not installed in machine.*
```
singularity run --app blast2120 /share/singularity/images/ccs/conda/amd-conda1-centos8.sinf
```
4. Ensure that the total length of mitochondrial sequences is less than 40kb. 
```
awk '{sum += $4} END {print sum}' MoMitochondrion.Pd8825.BLAST
```
## BLAST comparison against reference genome
1. Use blastn to run a BLAST search against the reference genome (B71). Use singularity if needed (see above). 
```
blastn -query B71v2sh_masked.fasta -subject Pd8825_97_2/Pd8825_final.fasta -evalue 1e-50 -max_target_seqs 20000 -outfmt '6 qseqid sseqid qstart qend sstart send btop' -out B71v2sh.Pd8825.BLAST
```

## Gene prediction

## Diagnostics 
*Due to an initial low BUSCO score of ~91%, the following are a series of tests to determine the cause of this high deviance from the reference genome.*
1. Examining the deviance from the reference genome within specific contigs in the Velvet assembly using BLAST (format 6).
```
blastn -query B71v2sh_masked.fasta -subject Pd8825_final.fasta -evalue 1e-50 -outfmt 6 -out B71v2sh.Pd8825.BLAST.6
sort -k1,1 -k7n B71v2sh.Pd8825.BLAST.6 | more
```
Singularity:
```
singularity run --app blast2120 /share/singularity/images/ccs/conda/amd-conda1-centos8.sinf 
```
From this BLAST, we generally found that large portions of the genome have low deviation. However, there were also substantial portions of high deviation (95-97% with contigs length >20,000 reads). This suggest that this sample may be a hybrid strain of *Pyricularia oryzae* and some other fungus. To identify this probable foreign DNA, we could manually identify some long contigs with high deviation, and run the following command (including singularity):
```
grep -A 1 Pd8825_contig5889 Pd8825_final.fasta | NR=2 awk '{print substr($1,1,14629)}' | singularity run --app blast2120 /share/singularity/images/ccs/conda/amd-conda1-centos8.sinf blastn -query - -db nr -outfmt 6 -remote
```
In this command, the contig we identified was Pd8825_contig5889. This BLAST primarily returned hits to *Pyricularia oryzae* so this command overall did not lead to very fruitful results. Alternatively, we would likely get more interesting results by identifying the Internal Transcribed Spacers (ITS) regions within this assmebly to BLAST instead. 

2. Assemble new genome with SPAdes.
The following SPAdes pipline trims, assmebles, and finalizes the genome assmebly. Thus the only input required are the raw reads and bash script.
```
sbatch trim-spades.sh . Pd8825 yes
```
***NOTE:** Bash script can be found under `SLURM_SCRIPTS`*

Results of SPAdes assembly (BUSCO command can be found above):
|  BUSCO score (complete)  | BUSCO score (complete + fragmented) | 
| ------------- | ------------- | 
| 98.4% | 98.6%  | 

This score is much higher! Now we want to know why this is...
Results of BLAST of SPAdes assembly against reference genome:
Manually observing the resulting alignments, we again find areas of high deviation and areas of low deviation.
Results of FastQC of SPAdes' trimmed reads:
|  Sequence  | Reads | Adaptor Content |
| ------------- | ------------- | -------------- |
| Trimmed Forward  | 7024576  | ~2% |
| Trimmed Reverse  | 7024576  | ~10% |

There is a relatively high percentage of adaptor content present in the reverse sequences. To investigate this, we will run a Velvet assembly on these trimmed reads. 

3. Velvet assembly with SPAdes' trimmed reads.

BUSCO score of resulting assembly:
|  BUSCO score (complete)  | BUSCO score (complete + fragmented) | 
| ------------- | ------------- | 
| 91.6% | 91.7%  | 

Since the BUSCO score of the Velvet assmebly with SPAdes' trimmed reads is also low, this suggests that it is SPAdes' assembly method that led to the difference in BUSCO score.

4. SPAdes assembly with initial manually trimmed reads.

BUSCO score of resulting assembly:
|  BUSCO score (complete)  | BUSCO score (complete + fragmented) | 
| ------------- | ------------- | 
| 98.3% | 98.5% | 
